src/routes/reservation/index.ts

```diff
@@ -0,0 +1,252 @@
+/* eslint-disable no-unused-vars */
+
+import * as awsx from '@pulumi/awsx';
+import { v4 as uuidv4 } from 'uuid';
+import { DynamoDB } from 'aws-sdk';
+import { marshall, parseBody, buildApiResponse, add500Handler, getUserId } from '$src/apiGatewayUtilities';
+import { query } from '$src/dynamodb/utils';
+import { createDynamo } from '$src/initAWS';
+
+function toResponse(entry: DynamoDB.AttributeMap) {
+  return {
+    id: entry.id.S,
+    propertyId: entry.propertyId.S,
+    customerId: entry.customerId.S,
+    bookedRoomsNumber: entry.bookedRoomsNumber.S,
+    beginDate: entry.beginDate.S,
+    endDate: entry.endDate.S,
+  };
+}
+
+function toArrayResponse(items: DynamoDB.Types.ItemList) {
+  return items.map((item) => toResponse(item));
+}
+
+export function getCustomerReservations() {
+  const dynamo = createDynamo();
+
+  const handler = async (event: awsx.apigateway.Request) => {
+    const customerId = getUserId(event);
+
+    const params = {
+      TableName: 'reservation',
+      KeyConditionExpression: 'customerId = :customerId',
+      ExpressionAttributeValues: {
+        ':customerId': { S: customerId },
+      },
+      Limit: 1000,
+    };
+
+    const reservations = await query(dynamo, params);
+
+    return buildApiResponse(200, toArrayResponse(reservations));
+  };
+
+  return add500Handler(handler);
+}
+
+function splitRange(beginDate: Date, endDate: Date) {
+  const dates = [];
+  const currDate = new Date(beginDate);
+
+  while (currDate < endDate){
+    dates.push(new Date(currDate));
+    currDate.setDate(currDate.getDate() + 1);
+  };
+
+  dates.push(new Date(endDate));
+
+  return dates;
+}
+
+function createGetLockedRoomsCountHandler(beginDate: Date, endDate: Date) {
+  return (items: DynamoDB.Types.ItemList) => {
+    const dates = splitRange(beginDate, endDate);
+
+    let totalLockedRoomsCount = 0;
+
+    for (const date of dates) {
+      const lockedRoomsCount = items.reduce((acc, item) => {
+        const beginItemDate = new Date(String(item.beginDate.S));
+        const endItemDate = new Date(String(item.endDate.S));
+
+        if (beginItemDate <= date && date <= endItemDate) {
+          return acc + Number(item.bookedRoomsNumber.N);
+        }
+
+        return acc;
+      }, 0);
+
+      if (lockedRoomsCount > totalLockedRoomsCount) {
+        totalLockedRoomsCount = lockedRoomsCount;
+      }
+    }
+
+    return totalLockedRoomsCount;
+  };
+}
+
+export function getAvailableCountReservations() {
+  const dynamo = createDynamo();
+
+  const handler = async (event: awsx.apigateway.Request) => {
+    const propertyId = event.pathParameters!.id;
+    const beginDate = new Date(event.queryStringParameters!.beginDate);
+    const endDate = new Date(event.queryStringParameters!.endDate);
+
+    if (beginDate < new Date()) {
+      return buildApiResponse(500, {
+        message: 'Reservation date unavailable',
+      });
+    }
+
+    const property = await dynamo.getItem({
```

Comment:
let's use the propertyMode.ts file, where the Property is properly statically typed